### PR TITLE
JUCX: Fixes for out-of-source build

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Generating pom file during configuration to support out of source build, removing env variable dependency.

## Why ?
To solve #3311 and got all needed information regarding paths from autoconf.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9433/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9455/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9501/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@petro-rudenko probably better use abs_top_builddir instead of top_builddir

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9531/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9545/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9551/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9557/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9568/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9572/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9577/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9618/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9619/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9628/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9631/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9637/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

File: bindings/java/pom.xml.in [hunk 1]
```diff
@@ -9,13 +9,20 @@
 
   <groupId>org.ucx</groupId>
   <artifactId>jucx</artifactId>
-  <version>0.0.1-SNAPSHOT</version>
+  <version>@VERSION@-SNAPSHOT</version>
```
Comment (yosefe) [inline bindings/java/pom.xml.in]: maybe remove `-SNAPSHOT` as well? keep just `@VERSION@`?

File: bindings/java/pom.xml.in [hunk 2]
```diff
@@ -26,13 +33,15 @@
...
-    <ucx.src.dir>${basedir}/../../src</ucx.src.dir>
-    <ucm.lib.path>${ucx.src.dir}/ucm/.libs</ucm.lib.path>
-    <ucs.lib.path>${ucx.src.dir}/ucs/.libs</ucs.lib.path>
-    <uct.lib.path>${ucx.src.dir}/uct/.libs</uct.lib.path>
-    <ucp.lib.path>${ucx.src.dir}/ucp/.libs</ucp.lib.path>
-    <ucx.inst.dir>${env.JUCX_INST}/lib</ucx.inst.dir>
+    <ucx.src.dir>@abs_top_srcdir@/src</ucx.src.dir>
+    <jucx.src.dir>${ucx.src.dir}/../bindings/java</jucx.src.dir>
+    <ucx.build.dir>@abs_top_builddir@</ucx.build.dir>
+    <native.dir>${ucx.build.dir}/bindings/java/src/main/native/</native.dir>
```
Comment (yosefe) [inline bindings/java/pom.xml.in]: can we remove trailing `/`?

File: bindings/java/pom.xml.in [hunk 3]
```diff
@@ -26,13 +33,15 @@
...
+    <ucm.lib.path>${ucx.build.dir}/src/ucm/.libs</ucm.lib.path>
+    <ucs.lib.path>${ucx.build.dir}/src/ucs/.libs</ucs.lib.path>
+    <uct.lib.path>${ucx.build.dir}/src/uct/.libs</uct.lib.path>
+    <ucp.lib.path>${ucx.build.dir}/src/ucp/.libs</ucp.lib.path>
+    <ucx.inst.dir>@prefix@/lib</ucx.inst.dir>
```
Comment (yosefe) [inline bindings/java/pom.xml.in]: `@libdir@` (ucx can be installed into /usr/lib64 for example)

File: bindings/java/pom.xml.in [hunk 4]
```diff
@@ -204,7 +216,7 @@
           </javahClassNames>
           <sources>
             <source>
-              <directory>${basedir}/src/main/native/</directory>
+              <directory>${jucx.src.dir}/src/main/native/</directory>
```
Comment (yosefe) [inline bindings/java/pom.xml.in]: can we remove trailing `/`?

File: bindings/java/src/main/native/Makefile.am [hunk 5]
```diff
@@ -5,7 +5,7 @@
 
 topdir=$(abs_top_builddir)
 javadir=$(top_srcdir)/bindings/java
-MVNCMD=$(MVN) -B -f $(javadir)/pom.xml -Dmaven.repo.local=$(topdir)/.deps/
+MVNCMD=$(MVN) -B -f $(topdir)/bindings/java/pom.xml -Dmaven.repo.local=$(topdir)/.deps/
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am]: can we remove trailing `/`?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9659/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
jenkins is quite flaky last time.

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9665/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9668/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9683/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9701/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9705/ for details (Mellanox internal link).

**Pull request merged**